crates/apub/src/api/search.rs

```diff
@@ -201,6 +201,12 @@ pub async fn search(
     }
   };
 
+  //filtering out removed and deleted comments from search results
+  let comments = comments
+    .into_iter()
+    .filter(|c| !c.comment.removed && !c.comment.deleted)
+    .collect();
```

Comment:
This isn't the right way to handle it either, filtering needs to be in SQL. 

In the comment query logic, you can check for `search_term` (since if that's provided, you know its a search), and filter out deleted and removed based on that.

Reply:
First try to add that [here](https://github.com/LemmyNet/lemmy/blob/main/crates/db_views/src/comment_view.rs#L219), into the existing `if let Some(search)`  . 

If that fails tho, your approach looks good also.